forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/api/registration/manual/ManualRegistrationApiController.java

```diff
@@ -172,15 +168,18 @@
     ) {
         Optional<Tpp> isTpp = tppStoreService.findByClientId(clientId);
         if (isTpp.isPresent()) {
-            tppRegistrationService.unregisterTpp(isTpp.get().getRegistrationResponse().getRegistrationAccessToken(), isTpp.get());
+            Tpp tpp = isTpp.get();
```

Comment:
Optional has an ifPresent method which accepts a consumer, exactly what the little block is doing. isPresent() followed by get() is a smell that is no different to doing a manual null check which is exactly what Optional is supposed to avoid.
```
tppStoreService.findByClientId(clientId).
  .ifPresent(tpp -> tppRegistrationService.unregisterTpp(tpp.getRegistrationResponse().getRegistrationAccessToken(),
            tpp));
```

Reply:
done